src/net/posix-stack.cc

```diff
@@ -918,7 +930,7 @@ posix_datagram_channel::receive() {
             }
         }
         return make_ready_future<datagram>(datagram(std::make_unique<posix_datagram>(
-            _recv._src_addr, dst, packet(fragment{_recv._buffer, size}, make_deleter([buf = _recv._buffer] { delete[] buf; })))));
+            _recv._src_addr, dst ? *dst : _address, packet(fragment{_recv._buffer, size}, make_deleter([buf = _recv._buffer] { delete[] buf; })))));
```

Comment:
I actually like the previous version with unconditional copying better (small copy = faster than conditional). But I don't think it matters, I'm sure it's impossible to measure a difference.